auth/grpctransport/grpctransport.go

```diff
@@ -341,7 +341,21 @@ func dial(ctx context.Context, secure bool, opts *Options) (*grpc.ClientConn, er
 			}),
 		)
 		// Attempt Direct Path
-		grpcOpts, transportCreds.Endpoint = configureDirectPath(grpcOpts, opts, transportCreds.Endpoint, creds)
+		var altsCreds *auth.Credentials
```

Comment:
Can all of this logic be moved inside `configureDirectPath`? This `dial` func is already quite noisy and complex.